Add ide tooling links to main guides (#14505)
quarkusio/quarkus: docs/src/main/asciidoc/ide-tooling.adoc

////
This guide is maintained in the main Quarkus repository
and pull requests should be submitted there:
https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
////
= Quarkus - Quarkus Tools in your favorite IDE

include::./attributes.adoc[]

[[project-creation]]
== IDE Overview

> The following IDE's has support for the community developed Quarkus Tools:

Reviewer: ```suggestion
The following IDE's have support for the community developed Quarkus Tools:
```

* https://marketplace.visualstudio.com/items?itemName=redhat.vscode-quarkus[Quarkus Tools for Visual Studio Code]
* https://marketplace.eclipse.org/content/quarkus-tools[Quarkus Tools for Eclipse]
* https://plugins.jetbrains.com/plugin/13234-quarkus/versions[Quarkus Tools for IntelliJ]

> * Quarkus Tools for Eclipse Che

Reviewer 2: ```suggestion
* https://github.com/eclipse/che-devfile-registry/blob/master/devfiles/quarkus/devfile.yaml[Eclipse Che devfile example using Quarkus Tools for Visual Studio Code]
```

In addition IntelliJ have additional support for Quarkus in their Ultimate non-open source version.

* https://www.jetbrains.com/help/idea/quarkus.html[IntelliJ Ultimate Edition built-in Quarkus support]

The table below gives an overview of the current IDE's with links and a high-level overview of their features.

:vscode-logo: https://simpleicons.org/icons/visualstudiocode.svg 
:eclipse-logo: https://simpleicons.org/icons/eclipseide.svg
:intellij-logo: https://simpleicons.org/icons/intellijidea.svg
:che-logo: che-icon-dark.svg
[cols="6*^", header]
|===
| .
| image:{vscode-logo}[200,200]
{empty} +
VSCode Quarkus Tools
| image:{eclipse-logo}[200,200]
{empty} +
Eclipse Quarkus Tools
| image:{intellij-logo}[100,100]
{empty} +
IntelliJ Quarkus Tools
| image:{intellij-logo}[100,100]
{empty} +
Intellij Ultimate
| image:{che-logo}[100,100]
{empty} +
Eclipse Che

|Description
|Visual Studio Code extension to install using the marketplace  
|Eclipse plugin to install into Eclipse using an updatesite
|Intellij plugin that works in Intellij Community and Ultimate. Available from Marketplace.
|Built-in Quarkus features available only in IntelliJ Ultimate

> |Built-in Quarkus features available in Eclipse Che incl. che.openshift.io.

Reviewer 2: ```suggestion
|Built-in Quarkus sample stacks available in Eclipse Che incl. online Che hosted by Red Hat. It uses the VS Code extension.
```

|Status
|Stable
|Alpha
|Beta
|Stable (since 2019.3)

> |Coming Soon!

Reviewer 2: :‑/

Author: this is from the original ide post made in 2019 - I didn't change these as I don't know what label you put on them today. Are they all just Stable?

Reviewer 2: We started packaging Che stable releases with a Quarkus devfile [since january 2019](https://github.com/eclipse/che-devfile-registry/pull/144).

Author: cool - but at the time I asked che pm for when and looked at che.openshift.io I was informed it was not available to use/consume. I've changed it to stable now.

Reviewer 2: You are right. I think we had resources problems on openshift.io at the time and the Quarkus sample didn't work at all.

|Downloads
| https://marketplace.visualstudio.com/items?itemName=redhat.vscode-quarkus[Marketplace]
{empty} +
 https://download.jboss.org/jbosstools/vscode/snapshots/vscode-quarkus/?C=M;O=D[Development Builds]
| https://download.jboss.org/jbosstools/photon/snapshots/builds/jbosstools-quarkus_master/[Development Update Site]
| https://plugins.jetbrains.com/plugin/13234-quarkus/versions[Marketplace]
{empty} +
https://download.jboss.org/jbosstools/intellij/snapshots/intellij-quarkus/[Development Builds]
| https://www.jetbrains.com/idea/nextversion/[Installer]

> |

Reviewer 2: ```suggestion
| https://che.openshift.io/f?url=https://raw.githubusercontent.com/redhat-developer/devfile/master/getting-started/quarkus/devfile.yaml[Start an example Che workspace online hosted by Red Hat]
```

|Source
|https://github.com/redhat-developer/vscode-quarkus[Github]
|https://github.com/jbosstools/jbosstools-quarkus[Github]
|https://github.com/redhat-developer/intellij-quarkus[Github]
|Closed-Source
|

|https://github.com/redhat-developer/quarkus-ls[Quarkus Language Server]
|icon:check[]
|icon:check[]
|icon:check[]
|icon:times[]
|icon:check[]

|Wizards w/code.quarkus.io
|icon:check[]
|icon:check[]
|https://issues.jboss.org/browse/JBIDE-26950[icon:times[]]
|icon:check[]
|icon:check[]

|Custom Wizard
|icon:times[]
|icon:times[]
|icon:check[]
|icon:check[]
|icon:times[]

> |Config editor

Reviewer 3: What do you mean with Config editor ? (syntax coloration?).

Author: editor that understands Quarkus properties. i.e. provide some quarkus specific content assist, validation, etc.

Reviewer 3: If I understand correctly, `Config editor = Config autocompletion + Config validation + ...` I though Config editor was a section which provide other things than auto completion, validation.

Author: ah sure - so yeah, so does the tool have any kind of quarkus config editing facility beyond just a text editor with syntax highlighting.

we might actually want to add one for yaml vs properties.

|icon:check[]
|icon:check[]
|icon:check[]
|icon:times[]
|icon:check[]

> 

Reviewer 3: Config definition (navigation from properties file to Java definition) could be good to add too.

Author: you mean seeing where inside quarkus extension the config is defined ?

Reviewer 3: Please see https://quarkus.io/blog/vscode-quarkus-1.1.0/#go-to-definition-support-for-config-properties-in-application-properties

Author: ack. added a line for it.

|Config autocompletion
|icon:check[]
|icon:check[]
|icon:check[]
|icon:check[]
|icon:check[]

|Config validation
|icon:check[]
|icon:check[]
|icon:check[]
|icon:check[]
|icon:check[]

|Config profiles
|icon:check[]
|icon:check[]
|icon:check[]
|icon:check[]
|icon:check[]

|Config outline 
|icon:check[]
|icon:check[]
|icon:check[]
|icon:times[]
|icon:check[]

|Easy Launch debug/dev:mode
|icon:check[]
|icon:times[]
|icon:check[]
|icon:check[]
|icon:check[]

|Quarkus Code Snippets
|icon:check[]

> |icon:times[]

Reviewer 3: As Quarkus Code Snippets are managed on LS server side all Quarkus toolings which are based on MP / Quarkus LS benefit with MicroProfile / Quarkus Code Snippets:

```
|Quarkus Code Snippets
|icon:check[]
|icon:check[]
|icon:check[]
|icon:times[]
|icon:check[]
```

|icon:times[]
|icon:times[]
|icon:check[]

> |Injection Discovery/Navigation

Reviewer 3: What do you mean about this feature? Could you explain more what is this feature please?

Author: intellij has features similar to what we had/have in jboss tools around find references like feature to navigate from a @Injection and similar annotations to possible sources and other way around.

Reviewer 3: Thanks @maxandersen for your clarification. Do you think LSP4MP should support this feature?

Author: if feasible - there are quite the special cases to cover and getting UX for it working would be important. in jboss tools very few realized it was even there so almost not used.

////
This guide is maintained in the main Quarkus repository
and pull requests should be submitted there:
https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
////
= Quarkus - Quarkus Tools in your favorite IDE

include::./attributes.adoc[]

[[project-creation]]
== IDE Overview

> The following IDE's has support for the community developed Quarkus Tools:

Reviewer 4: ```suggestion
The following IDEs have support for the community developed Quarkus Tools:
```
